sdk/storage/src/common/storage_error.cpp

```diff
@@ -5,21 +5,30 @@
 
 #include "common/constants.hpp"
 #include "common/xml_wrapper.hpp"
+#include "http/policy.hpp"
 #include "json.hpp"
 
 #include <type_traits>
 
 namespace Azure { namespace Storage {
   StorageError StorageError::CreateFromResponse(
+      Azure::Core::Context context,
       std::unique_ptr<Azure::Core::Http::RawResponse> response)
   {
-    auto bodyStream = response->GetBodyStream();
     std::vector<uint8_t> bodyBuffer;
-    if (bodyStream != nullptr)
+    bool useBodyStream = context.HasKey(Azure::Core::Http::Details::c_GetStreamForBody)
```

Comment:
Maybe we can expect all fail requests to return with vectorBody and never with bodyStream.
@JeffreyRichter if we perform a request for download (or any api with context option to return bodyStream) and if fails (code >= 300). Can we assume pipeline would download body directly?
Then convince layer doesn't need to check if there is a bodyStream or not like here.
What do you think?

Reply:
This is a reasonable feature.